kapre/time_frequency.py

```diff
@@ -452,3 +452,323 @@ def get_config(self):
+          (new_rows, new_cols, filters)` if `data_format='channels_last'`.  `rows`
+          and `cols` values might have changed due to padding.
+
+    Returns:
+        A tensor of rank 4+ representing
+        `activation(conv2d(inputs, kernel) + bias)`.
+
+    Raises:
+        ValueError: if `padding` is `"causal"`.
+        ValueError: when both `strides > 1` and `dilation_rate > 1`.
+        """
+    def __init__(self,
+               filters,
+               kernel_size,
+               strides=(1, 1),
+               padding='valid',
+               data_format=None,
+               dilation_rate=(1, 1),
+               groups=1,
+               activation=None,
+               use_bias=True,
+               kernel_initializer='glorot_uniform',
+               bias_initializer='zeros',
+               kernel_regularizer=None,
+               bias_regularizer=None,
+               activity_regularizer=None,
+               kernel_constraint=None,
+               bias_constraint=None,
+               frequency_first=True,
+               **kwargs):
+        if groups > 1:
+            warnings.warn(
+                'Grouped Conv2D are not supported.'
+                'Only the first group will be frequency aware.'
+            )
+        self.frequency_first=frequency_first
+        super(FreqAwareConv, self).__init__(
+            filters=filters,
+            kernel_size=kernel_size,
+            strides=strides,
+            padding=padding,
+            data_format=data_format,
+            dilation_rate=dilation_rate,
+            groups=groups,
+            activation=activation,
+            use_bias=use_bias,
+            kernel_initializer=kernel_initializer,
+            bias_initializer=bias_initializer,
+            kernel_regularizer=kernel_regularizer,
+            bias_regularizer=bias_regularizer,
+            activity_regularizer=activity_regularizer,
+            kernel_constraint=kernel_constraint,
+            bias_constraint=bias_constraint,
+            **kwargs)
+
+
+    def call(self, inputs):
+        inputs = self._add_freq_info_channel(inputs)
+        outputs = super(FreqAwareConv, self).call(inputs)
+        return outputs
+
+    def _add_freq_info_channel(self, inputs):
+        # adapted from https://github.com/tchaton/CoordConv-keras/blob/master/coord.py
+        shape = tf.shape(inputs)
+        batch_size_tensor, x_dim, y_dim, c = shape[0], shape[1], shape[2], shape[3]
+        if not self.frequency_first:
+             x_dim, y_dim =  y_dim, x_dim
+        xx_ones = tf.ones([batch_size_tensor, 1], dtype=tf.float32)
+        
+        xx_ones = tf.expand_dims(xx_ones, axis=-1)
+        
+        xx_range = tf.tile(tf.expand_dims(tf.range(x_dim), 0), [batch_size_tensor, 1])
+        xx_range = tf.cast(xx_range, tf.float32)
+        xx_range = tf.expand_dims(xx_range, axis=1)
+        
+        xx_channel = tf.matmul(xx_ones, xx_range)
```

Comment:
that's true. my original implementation is online pytorch: https://github.com/kkoutini/cpjku_dcase19/blob/937f90af6eff668f38e4a84c4e90b2acb218a4c7/models/cp_faresnet.py#L71
but since I never used keras before I relied on the implementation of CordConv https://github.com/tchaton/CoordConv-keras/blob/master/coord.py

Reply:
I'll try to make it clearer in the coming days